File: test_extension.ts
// Test TypeScript file for AI-TD Detector Extension
// Contains patterns that should trigger AI-TD detection

interface DataItem {
    type: string;
    value: number;
    metadata: Record<string, any>;
    priority?: number;
    status?: string;
    valid?: boolean;
    active?: boolean;
}

interface ProcessingOptions {
    threshold?: number;
    mode?: 'auto' | 'manual';
    debug?: boolean;
    outputFormat?: string;
    validate?: boolean;
    maxIterations?: number;
    tolerance?: number;
}

interface ProcessedItem {
    original: number;
    processed: number;
    metadata: Record<string, any>;
    timestamp: Date;
}

// Complex function with high cyclomatic complexity
function complexDataProcessor(data: DataItem[], options: ProcessingOptions = {}): ProcessedItem[] | null {
    const {
        threshold = 0.5,
        mode = 'auto',
        debug = false,
        outputFormat = 'json',
        validate = true,
        maxIterations = 1000,
        tolerance = 0.001
    } = options;

    if (!data || data.length === 0) {
        return null;
    }

    const results: ProcessedItem[] = [];
    
    for (let i = 0; i < data.length; i++) {
        if (i > maxIterations) break;
        
        const item = data[i];
        
        // Complex nested conditional logic
        if (item.type === 'A') {
            if (item.value > threshold) {
                if (mode === 'auto') {
                    const processed = processTypeA(item, debug);
                    results.push(processed);
                } else if (mode === 'manual') {
                    const processed = manualProcess(item, outputFormat);
                    results.push(processed);
                } else {
                    const processed = defaultProcess(item);
                    results.push(processed);
                }
            }
        } else if (item.type === 'B') {
            if (item.status === 'active') {
                if (validate) {
                    if (validateItem(item)) {
                        const processed = processTypeB(item, debug);
                        results.push(processed);
                    }
                } else {
                    const processed = processTypeB(item, debug);
                    results.push(processed);
                }
            }
        } else if (item.type === 'C') {
            // Duplicated logic pattern
            if (item.priority && item.priority > 5) {
                if (mode === 'auto') {
                    const processed = processTypeC(item, debug);
                    results.push(processed);
                } else if (mode === 'manual') {
                    const processed = manualProcessC(item, outputFormat);
                    results.push(processed);
                } else {
                    const processed = defaultProcessC(item);
                    results.push(processed);
                }
            }
        }
    }
    
    return results;
}

// Function without proper error handling
function processTypeA(item: DataItem, debug: boolean): ProcessedItem {
    // No error handling for missing properties
    const value = item.value * 2;
    const metadata = item.metadata;
    
    // Complex calculation
    if (value > 100) {
        return {
            original: value,
            processed: value / 2,
            metadata: metadata,
            timestamp: new Date()
        };
    } else {
        return {
            original: value,
            processed: value * 1.5,
            metadata: metadata,
            timestamp: new Date()
        };
    }
}

// Duplicated function - should trigger duplication detection
function processTypeB(item: DataItem, debug: boolean): ProcessedItem {
    // No error handling for missing properties
    const value = item.value * 2;
    const metadata = item.metadata;
    
    // Duplicated logic from processTypeA
    if (value > 100) {
        return {
            original: value,
            processed: value / 2,
            metadata: metadata,
            timestamp: new Date()
        };
    } else {
        return {
            original: value,
            processed: value * 1.5,
            metadata: metadata,
            timestamp: new Date()
        };
    }
}

// Another duplicated function
function processTypeC(item: DataItem, debug: boolean): ProcessedItem {
    // No error handling for missing properties
    const value = item.value * 2;
    const metadata = item.metadata;
    
    // Again duplicated logic
    if (value > 100) {
        return {
            original: value,
            processed: value / 2,
            metadata: metadata,
            timestamp: new Date()
        };
    } else {
        return {
            original: value,
            processed: value * 1.5,
            metadata: metadata,
            timestamp: new Date()
        };
    }
}

// Function with poor documentation
function manualProcess(item: DataItem, outputFormat: string): DataItem {
    // Missing proper JSDoc
    return item;
}

// Another poorly documented function
function manualProcessC(item: DataItem, outputFormat: string): DataItem {
    // Missing proper JSDoc
    return item;
}

function defaultProcess(item: DataItem): DataItem {
    return item;
}

function defaultProcessC(item: DataItem): DataItem {
    return item;
}

// Validation function without error handling
function validateItem(item: DataItem): boolean {
    // No error handling for undefined properties
    return (item.valid || false) && (item.active || false);
}

// Undocumented function
function undocumentedFunction(x: number, y: number): number {
    return x + y;
}

// Complex class with multiple issues
class DataProcessor {
    private config: ProcessingOptions;
    private processed: ProcessedItem[];

    constructor(config: ProcessingOptions) {
        this.config = config;
        this.processed = [];
    }
    
    // Complex method with high complexity
    processBatch(data: DataItem[]): ProcessedItem[] {
        // No error handling
        for (let i = 0; i < data.length; i++) {
            const item = data[i];
            
            if (item.type === 'A') {
                if (item.value > (this.config.threshold || 0.5)) {
                    if (this.config.mode === 'auto') {
                        this.processed.push(this.processItem(item));
                    } else if (this.config.mode === 'manual') {
                        this.processed.push(this.manualItem(item));
                    }
                }
            } else if (item.type === 'B') {
                if (item.status === 'active') {
                    this.processed.push(this.processItem(item));
                }
            }
        }
        
        return this.processed;
    }
    
    // Method without documentation
    processItem(item: DataItem): ProcessedItem {
        return {
            original: item.value,
            processed: item.value * 2,
            metadata: item.metadata,
            timestamp: new Date()
        };
    }
    
    // Another undocumented method
    manualItem(item: DataItem): ProcessedItem {
        return {
            original: item.value,
            processed: item.value * 1.5,
            metadata: item.metadata,
            timestamp: new Date()
        };
    }
}

// Main execution without error handling
function main(): void {
    // Test data
    const testData: DataItem[] = [
        { type: 'A', value: 150, metadata: { source: 'test' }, priority: 8 },
        { type: 'B', value: 75, metadata: { source: 'test' }, status: 'active' },
        { type: 'C', value: 200, metadata: { source: 'test' }, priority: 10 }
    ];
    
    // No try-catch or error handling
    const results = complexDataProcessor(testData);
    console.log(`Processed ${results?.length || 0} items`);
    
    // No error handling for file operations
    const fs = require('fs');
    fs.writeFileSync('output.json', JSON.stringify(results, null, 2));
}

// Execute main function
main();

File: test_extension.py
# Test file for AI-TD Detector Extension
# This file contains various patterns that should trigger AI-TD detection

import os
import sys
import json
from typing import List, Dict, Optional

def complex_function(data: List[Dict], threshold: float = 0.5, 
                    mode: str = "auto", debug: bool = False, 
                    output_format: str = "json", validate: bool = True,
                    max_iterations: int = 1000, tolerance: float = 0.001):
    """This function is intentionally complex to trigger complexity detection."""
    if not data:
        return None
    
    result = []
    for i, item in enumerate(data):
        if i > max_iterations:
            break
            
        # Complex nested logic
        if item.get('type') == 'A':
            if item.get('value') > threshold:
                if mode == "auto":
                    processed = process_type_a(item, debug)
                elif mode == "manual":
                    processed = manual_process(item, output_format)
                else:
                    processed = default_process(item)
                result.append(processed)
        elif item.get('type') == 'B':
            if item.get('status') == 'active':
                if validate:
                    if validate_item(item):
                        result.append(process_type_b(item, debug))
                else:
                    result.append(process_type_b(item, debug))
        elif item.get('type') == 'C':
            # Similar logic repeated (duplication)
            if item.get('priority') > 5:
                if mode == "auto":
                    processed = process_type_c(item, debug)
                elif mode == "manual":
                    processed = manual_process_c(item, output_format)
                else:
                    processed = default_process_c(item)
                result.append(processed)
    
    return result

def process_type_a(item: Dict, debug: bool) -> Dict:
    """Process type A items."""
    # No error handling - should trigger error handling detection
    value = item['value'] * 2
    metadata = item['metadata']
    
    # Complex calculation
    if value > 100:
        result = {
            'original': value,
            'processed': value / 2,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    else:
        result = {
            'original': value,
            'processed': value * 1.5,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    
    return result

def process_type_b(item: Dict, debug: bool) -> Dict:
    """Process type B items - similar to type A (duplication)."""
    # No error handling - should trigger error handling detection
    value = item['value'] * 2
    metadata = item['metadata']
    
    # Similar complex calculation (duplication)
    if value > 100:
        result = {
            'original': value,
            'processed': value / 2,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    else:
        result = {
            'original': value,
            'processed': value * 1.5,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    
    return result

def process_type_c(item: Dict, debug: bool) -> Dict:
    """Process type C items - another duplicated function."""
    # No error handling - should trigger error handling detection
    value = item['value'] * 2
    metadata = item['metadata']
    
    # Again similar logic (duplication)
    if value > 100:
        result = {
            'original': value,
            'processed': value / 2,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    else:
        result = {
            'original': value,
            'processed': value * 1.5,
            'metadata': metadata,
            'timestamp': os.path.getmtime(__file__)
        }
    
    return result

def manual_process(item: Dict, output_format: str) -> Dict:
    """Manual processing without documentation."""
    # Missing docstring details - should trigger documentation detection
    return item

def manual_process_c(item: Dict, output_format: str) -> Dict:
    """Manual processing for type C without documentation."""
    # Missing docstring details - should trigger documentation detection
    return item

def default_process(item: Dict) -> Dict:
    """Default processing."""
    return item

def default_process_c(item: Dict) -> Dict:
    """Default processing for type C."""
    return item

def validate_item(item: Dict) -> bool:
    """Validate item without error handling."""
    # No error handling for missing keys
    return item['valid'] and item['active']

# Missing documentation for this function
def undocumented_function(x, y):
    return x + y

# Main execution without error handling
if __name__ == "__main__":
    # Test data that should trigger various detections
    test_data = [
        {'type': 'A', 'value': 150, 'metadata': {'source': 'test'}, 'priority': 8},
        {'type': 'B', 'value': 75, 'metadata': {'source': 'test'}, 'status': 'active'},
        {'type': 'C', 'value': 200, 'metadata': {'source': 'test'}, 'priority': 10}
    ]
    
    # No error handling for the main processing
    results = complex_function(test_data)
    print(f"Processed {len(results)} items")
